### PR TITLE
fixed creation process of JWT token

### PR DESCRIPTION
**Pull request by Kar977**

### CODE REVIEW

File: auth/auth0_client.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+    :param request: The HTTP request object.
+    :return: A message confirming the user has logged out.
+    """
+
+    request.session.clear()
+
+    return {"message": "logged out successfully"}
+
+# ToDO delete it after implementation of tests
```
Comment (kgarbacinski) [inline auth/auth0_client.py:133]: redundant?

File: auth/auth0_client.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    5. Returning the token's subject (sub) and permissions (if present) in a structured TokenData object.
+
+    :param token: The JWT token string to verify.
+    :return: TokenData object containing 'sub' (subject) and optional 'permissions'.
+    :raises: HTTPException if the token is invalid or expired.
+    """
+    try:
+        jwks = get_auth0_public_key()
+        unverified_header = jwt.get_unverified_header(token)
+        print("unverified_header = ", unverified_header)
```
Comment (kgarbacinski) [inline auth/auth0_client.py]: why all these prints here?

File: auth/auth0_client.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+from fastapi import APIRouter, Request, Depends, HTTPException
+from fastapi.responses import RedirectResponse
+from jose import jwt, JWTError
+from main import oauth
+from settings import Settings
+from starlette.exceptions import HTTPException
+
+load_dotenv()
+
+AUTH0_DOMAIN = os.getenv("AUTH0_DOMAIN")
```
Comment (kgarbacinski) [inline auth/auth0_client.py]: move it to external file / settings.py / external class

File: auth/__pycache__/schemas.cpython-311.pyc
Comment (kgarbacinski) [inline auth/__pycache__/schemas.cpython-311.pyc:1]: all these .pyc should be placed in a .gitignore file

File: auth/auth0_client.py (hunk 1 above)
Comment (Kar977) [reply auth/auth0_client.py:133]: deleted

File: auth/auth0_client.py (hunk 2 above)
Comment (Kar977) [reply auth/auth0_client.py]: prints deleted - scraps from debugging

File: auth/auth0_client.py (hunk 3 above)
Comment (Kar977) [reply auth/auth0_client.py]: moved to external file settings.py and class Settings where crucial constants are taken from .env

File: routers/customer_manager/workday.py [hunk 4]
```diff
@@ -8,7 +8,10 @@
 MICROSERVICE_URL = "http://localhost:8001"
 
 @router.post("/workdays/workday/")
-async def create_workday(workday_request: CreateWorkdayRequest):
+async def create_workday(workday_request: CreateWorkdayRequest, request: Request):
+    from auth.auth0_client import get_current_user as check_if_logged
```
Comment (kgarbacinski) [inline routers/customer_manager/workday.py:12]: this import shouldn't be here; if you're getting a circular import error, then move the invalid function somewhere else (like to other file)

File: auth/auth0_client.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+    :param request: The HTTP request object.
+    :return: A message confirming the user has logged out.
+    """
+
+    request.session.clear()
+
+
+# ToDO delete it after implementation of tests
+# End point only for check if security with token works correctly
```
Comment (kgarbacinski) [inline auth/auth0_client.py:134]: all comments should be remove, there are still a few

File: auth/auth0_client.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        for key in jwks:
+            if key["kid"] == unverified_header["kid"]:
+                rsa_key = {
+                    "kty": key["kty"],
+                    "kid": key["kid"],
+                    "use": key["use"],
+                    "n": key["n"],
+                    "e": key["e"]
+                }
+        if rsa_key:
```
Comment (kgarbacinski) [inline auth/auth0_client.py:58]: may it throw an error? should it be inside the try block or moved up?

**Pull request closed without merging**